.github/copilot-instructions.md

```diff
@@ -0,0 +1,165 @@
+# Orionrobots Website
+
+**ALWAYS** reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+Orionrobots is a static website about robotics using Eleventy (11ty) static site generator with Webpack for asset bundling. The site includes a blog with Jekyll-style posts, wiki pages, and robotics tutorials. It's hosted on 3rd party hosting with Apache configuration.
+
+## Working Effectively
+
+### CRITICAL: Bootstrap and Build Commands
+- Bootstrap and run the repository:
+  - `docker compose up` -- **NEVER CANCEL** - builds and starts development server. Set timeout to 300+ seconds for initial build.
+- Build assets and site manually:
+  - `docker compose run dist` -- builds webpack bundle.js, takes ~30 seconds
+  - `docker compose run build` -- builds full static site, takes ~3 minutes. Set timeout to 240+ seconds.
+- Alternative native workflow:
+  ```bash
+  npm install
+  npm run dist
+  npm run 11ty
+  ```
+
+### Development Server
+- Run local development server:
+  - `docker compose up serve` -- starts Eleventy dev server on port 8081, **NEVER CANCEL** - builds then watches files. Set timeout to 300+ seconds for initial build.
+  - Alternative native: `npm run serve` -- starts Eleventy dev server directly
+  - Manual serving: After building, serve with Python: `cd _site && python3 -m http.server 8082`
+
+### Testing Commands
+- BDD integration tests:
+  - `docker compose run test` -- runs Cucumber.js tests with Playwright in containerized environment
+  - Alternative native: `npm run test:bdd` or `npm test` -- runs tests directly on host
+  - **Note**: Playwright installation may fail in some CI environments
+  - Tests require a running staging server to test against
+
+## Validation Scenarios
+
+### Manual Validation Required
+**ALWAYS** manually validate any new code by testing these scenarios after making changes:
+1. **Homepage Loading**: Visit the root page and verify navigation works
+2. **Blog Post Access**: Navigate to a recent blog post and verify images load
+3. **Wiki Navigation**: Test wiki links and cross-references work
+4. **Search Functionality**: Test any search features if they exist
+
+### Staging Test Validation
+Always test these key pages that are verified in CI:
+- Homepage: `/` (should return 200 OK)
+- Construction guide: `/construction_guide.html` (should return 200 OK)
+- Wiki pages: `/wiki/lego` and similar (check for 404s)
+
+### Build Verification
+- Run `docker compose run dist && docker compose run build` and verify `_site` directory is created with content
+- Alternative native: `npm run dist && npm run 11ty`
+- Check that `_site/index.html` exists and contains proper HTML structure
+- Verify `dist/bundle.js` exists and is approximately 330KB
```

Comment:
While it should exist, I'm not sure about size verification. One goal would be to reduce this bundle at some point. Perhaps for now this should be non-zero, at least until we can get playwright to verify more?